src/EnergyPlus/SteamCoils.cc

```diff
@@ -337,43 +337,40 @@ namespace SteamCoils {
                                                                                           NodeInputManager::CompFluidStream::Primary,
                                                                                           ObjectIsNotParent);
 
-            {
-                auto const SELECT_CASE_var(UtilityRoutines::MakeUPPERCase(AlphArray(7)));
-                // TEMPERATURE SETPOINT CONTROL or ZONE LOAD CONTROLLED Coils
-                if (SELECT_CASE_var == "TEMPERATURESETPOINTCONTROL") {
-                    state.dataSteamCoils->SteamCoil(CoilNum).TypeOfCoil = TemperatureSetPointControl;
-                    state.dataSteamCoils->SteamCoil(CoilNum).TempSetPointNodeNum =
-                        GetOnlySingleNode(state,
-                                          AlphArray(8),
-                                          ErrorsFound,
-                                          DataLoopNode::ConnectionObjectType::CoilHeatingSteam,
-                                          AlphArray(1),
-                                          DataLoopNode::NodeFluidType::Air,
-                                          DataLoopNode::ConnectionType::Sensor,
-                                          NodeInputManager::CompFluidStream::Primary,
-                                          ObjectIsNotParent);
-                    if (state.dataSteamCoils->SteamCoil(CoilNum).TempSetPointNodeNum == 0) {
-                        ShowSevereError(state,
-                                        std::string{RoutineName} + cAlphaFields(8) + " not found for " + CurrentModuleObject + " = " + AlphArray(1));
-                        ShowContinueError(state, "..required for Temperature Setpoint Controlled Coils.");
-                        ErrorsFound = true;
-                    }
-
-                } else if (SELECT_CASE_var == "ZONELOADCONTROL") {
-                    state.dataSteamCoils->SteamCoil(CoilNum).TypeOfCoil = ZoneLoadControl;
-
-                    if (!lAlphaBlanks(8)) {
-                        ShowWarningError(state, std::string{RoutineName} + "ZoneLoad Controlled Coil, so " + cAlphaFields(8) + " not needed");
-                        ShowContinueError(state, "for " + CurrentModuleObject + " = " + AlphArray(1));
-                        state.dataSteamCoils->SteamCoil(CoilNum).TempSetPointNodeNum = 0;
-                    }
-
-                } else {
+            auto const controlMode(UtilityRoutines::MakeUPPERCase(AlphArray(7)));
```

Comment:
There is no point to this change. If you are going to leave the pattern for now anyway, just leave the `SELECT_CASE_var` name so at least it will be easy to find later.

Reply:
I cleaned this up properly, making an enum and a matching std::array, using getEnumerationValue, and changing it to a switch block.  In the process, I realized that one of the integer types was never being used, so cleaned up a few things in the header and code accordingly.  Should be much better now.